src/application.ts

```diff
@@ -25,8 +25,8 @@ export class Application {
   public catchErrors?: boolean
   public log: LoggerWithTarget
 
-  constructor (options: Options) {
-    const opts = options || {}
+  constructor (options?: Options) {
+    const opts = options || {} as any
```

Comment:
I'm not 💯clear here. Should it be possible to do `new Application()` with no parameters? Right now `app`, `cache`, and `catchErrors` are not defined if we don't pass anything in. Router is defined as optional because it can be `opts.router || express.Router()`, but the others don't have a fallback value.